File: PersianCalendar/src/main/java/com/byagowi/persiancalendar/di/dependencies/AppDependency.java
package com.byagowi.persiancalendar.di.dependencies;

import android.content.SharedPreferences;
import android.preference.PreferenceManager;

import com.byagowi.persiancalendar.MainApplication;

import javax.inject.Inject;
import javax.inject.Singleton;

@Singleton
public final class AppDependency {
    private final SharedPreferences sharedPreferences;

    @Inject
    public AppDependency(MainApplication app) {
        sharedPreferences = PreferenceManager.getDefaultSharedPreferences(app);
    }

    public SharedPreferences getSharedPreferences() {
        return sharedPreferences;
    }
}

File: PersianCalendar/src/main/java/com/byagowi/persiancalendar/di/dependencies/AppDependency.kt
package com.byagowi.persiancalendar.di.dependencies

import android.content.SharedPreferences
import android.preference.PreferenceManager

import com.byagowi.persiancalendar.MainApplication

import javax.inject.Inject
import javax.inject.Singleton

@Singleton
class AppDependency @Inject
constructor(app: MainApplication) {
    val sharedPreferences: SharedPreferences = PreferenceManager.getDefaultSharedPreferences(app)
}
